sysfs/system_cpu.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sysfs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/procfs/internal/util"
+)
+
+// SystemCPUCpufreq contains stats from devices/system/cpu/cpu[0-9]*/cpufreq/...
+type SystemCPUCpufreq struct {
+	Name               string
+	CurrentFrequency   uint64
+	MinimumFrequency   uint64
+	MaximumFrequency   uint64
+	TransitionLatency  uint64
+	AvailableGovernors string
+	Driver             string
+	Govenor            string
+	RelatedCpus        string
+	SetSpeed           string
+}
+
+// SystemCpufreq is a collection of SystemCPUCpufreq for every CPU.
+type SystemCpufreq []SystemCPUCpufreq
```

Comment:
Since this type has no methods, I would probably remove it and just return the slice directly.

Reply:
It does now. `func (s SystemCpufreq) parseCpufreqCpuinfo(...)`

Reply:
I think @mdlayher means getting rid of `SystemCpufreq` and returning `[]SystemCPUCpufreq` instead and I agree on that.